### PR TITLE
mongoose schemas

### PR DESCRIPTION
**Pull request by deepanshu2506**
# Description

Added the mongoose schemas to start development

- userProfile Schema 
contains validations for age above 13

- posts Schema
contains a complex recursive object to map comments on the post

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,13 +1,29 @@
...
+
 app.use(bodyParser.urlencoded({ extended: false }));
 app.use(bodyParser.json());
 
 app.use(routes);
 
-const port = process.env.PORT || 8080;
+mongoose.connect(
+  config.dbUrl,
+  { useNewUrlParser: true, useUnifiedTopology: true },
```
Comment (amandesai01) [inline app.js:17]: Please move config.dbUrl to `dbSecrets.js` because it will be different for different people.

Comment (amandesai01) [reply app.js:17]: I have added that file in `.gitignore`

Comment (deepanshu2506) [reply app.js:17]: Not needed

Comment (deepanshu2506) [reply app.js:17]: No need to put it in dbSecrets.just create a db with the same name. It will help in the long run

**Pull request merged**